app/controllers/application_controller.rb

```diff
@@ -2,4 +2,24 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  private
+
+  def current_user
+    @_current_user ||= session[:user_id]
```

Comment:
@ltw why do you think it will always be run? It's conditionally setting @_current_user (maybe a bad name, but still valid).